src/post/Write/Write.js

```diff
@@ -102,17 +113,42 @@ class Write extends React.Component {
         initialBody: draftPost.body || '',
         initialReward: draftPost.reward || '50',
         initialUpvote: draftPost.upvote,
-        isUpdating: isUpdating || false,
+        initialUpdatedDate: draftPost.lastUpdated || Date.now(),
+        isUpdating: draftPost.isUpdating || false,
+        didExist: true,
       });
     }
   }
 
+  onCancel = () => {
+    if (this.state.didExist) {
+      // If the draft Existed return it to previous state
+      const initalForm = {
+        body: this.state.initialBody,
+        title: this.state.initialTitle,
+        reward: this.state.initialReward,
+        upvote: this.state.initialUpvote,
+        lastUpdated: this.state.initialUpdatedDate,
+        topics: this.state.initialTopics,
+      };
+
+      this.props
+        .saveDraft({ postData: this.getNewPostData(initalForm), id: this.props.draftId })
+        .then(() => {
+          this.props.goBack();
+        });
+    } else {
+      // If didn't exist just delete it.
+      this.props.deleteDraft(this.props.draftId).then(() => {
+        this.props.replace('/');
```

Comment:
Not sure how to get the previous state and replacing the current route with last one.

Also, in case the draft was saved (got id) it should not only replace the current one but the last two("/write","/write?draft=id").